src/us/kbase/workspace/database/provenance/Contributor.java

```diff
@@ -0,0 +1,314 @@
+		PERSON, ORGANIZATION;
+
+		/**
+		 * Get a contributor type based on a supplied string.
+		 *
+		 * @param contributorType
+		 *                 a string representing a contributor type.
+		 * @return a contributor type.
+		 * @throws IllegalArgumentException
+		 *                 if there is no contributor type
+		 *                 related to the input string.
+		 */
+		public static ContributorType getType(final String contributorType) {
+			final String lowercaseInput = Util.checkString(contributorType, "contributorType").toLowerCase();
+			switch (lowercaseInput) {
+				case "person":
+					return ContributorType.PERSON;
+				case "organization":
+				case "organisation":
+					return ContributorType.ORGANIZATION;
+				default:
+					throw new IllegalArgumentException("Invalid contributorType: " + contributorType);
+			}
+		}
+	}
+
+	private final ContributorType contributorType;
+	private final String contributorID;
+	private final String name;
+	private final String creditName;
+	private final List<Organization> affiliations;
+	private final List<ContributorRole> contributorRoles;
+
+	private Contributor(
+			final ContributorType contributorType,
+			final String name,
+			final String creditName,
+			final String contributorID,
+			final List<Organization> affiliations,
+			final List<ContributorRole> contributorRoles) {
+		this.contributorType = contributorType;
+		this.name = name;
+		this.creditName = creditName;
+		this.contributorID = contributorID;
+		this.affiliations = affiliations;
+		this.contributorRoles = contributorRoles;
+	}
+
+	/**
+	 * Gets the contributor type, either person or organization.
+	 *
+	 * @return the contributor type
+	 */
+	public ContributorType getContributorType() {
+		return contributorType;
+	}
+
+	/**
+	 * Gets the name associated with the contributor.
+	 *
+	 * @return the contributor's name
+	 */
+	public String getName() {
+		return name;
+	}
+
+	/**
+	 * Gets the credit name associated with the contributor.
+	 *
+	 * @return the contributor's credit name, if present
+	 */
+	public Optional<String> getCreditName() {
+		return Optional.ofNullable(creditName);
+	}
+
+	/**
+	 * Gets the ID for the contributor.
+	 *
+	 * @return the contributor ID, if present
+	 */
+	public Optional<String> getContributorID() {
+		return Optional.ofNullable(contributorID);
+	}
+
+	/**
+	 * Gets the contributor's affiliations.
+	 *
+	 * @return the contributor's affiliations, if present.
+	 */
+	public List<Organization> getAffiliations() {
+		return Common.getList(affiliations);
+	}
+
+	/**
+	 * Gets the roles that the contributor filled.
+	 *
+	 * @return the contributor roles, if present.
+	 */
+	public List<ContributorRole> getContributorRoles() {
+		return Common.getList(contributorRoles);
+	}
+
+	/**
+	 * Gets the roles that the contributor filled.
+	 *
+	 * @return the contributor roles, if present.
+	 */
+	public List<String> getContributorRoleStrings() {
+		return Common.immutable(
```

Comment:
Strictly speaking since this method is returning a new list, it's ok that it's not immutable. Returning references to mutable class instances is bad, obviously.

Fine to match the behavior of the other methods though

Reply:
:+1: